tests/unit/handlers/test_mongodb.py

```diff
@@ -227,31 +224,348 @@ def test_get_tables(self):
+        self.assertEqual(df["Field"].tolist(), ["_id", "name", "plot", "runtime"])
+        self.assertEqual(df["Type"].tolist(), ["str", "str", "str", "int"])
+
+    # use subquery for select
+    def test_query_select_with_subquery_success(self):
+        """
+        Test if the `query` method returns a response object with a data frame containing the query result for a select with subquery.
+        e.g., SELECT * FROM (SELECT * FROM theaters);
+        """
+        self.mock_connect.return_value[self.dummy_connection_data["database"]].list_collection_names.return_value = [
+            "movies",
+            "theaters",
+        ]
+
+        self.mock_connect.return_value[self.dummy_connection_data["database"]]["theaters"].aggregate.return_value = [
+            {
+                "_id": ObjectId("5f5b3f3b3f3b3f3b3f3b3f3b"),
+                "name": "Cinema City",
+                "location": "Downtown",
+            }
+        ]
+
+        subquery = ast.Select(
+            targets=[
+                Star(),
+            ],
+            from_table=ast.Identifier("theaters"),
+        )
+
+        main_query = ast.Select(
+            targets=[
+                Star(),
+            ],
+            from_table=subquery,
+        )
+
+        response = self.handler.query(main_query)
+
+        assert isinstance(response, Response)
+        self.assertEqual(response.type, RESPONSE_TYPE.TABLE)
+
+        df = response.data_frame
+        self.assertEqual(len(df), 1)
+        self.assertEqual(df.columns.tolist(), ["_id", "name", "location"])
+        self.assertEqual(df["name"].tolist(), ["Cinema City"])
+
+    def test_query_select_with_complex_subquery_success(self):
+        """
+        Test if the `query` method returns a response object with a data frame containing the query result for a select with complex subquery.
+        e.g. SELECT * FROM (SELECT CAST(customer_id AS VARCHAR) AS cust_id, CAST(first_name AS VARCHAR) AS fname FROM mongo_db.customers)
+        """
+        self.mock_connect.return_value[self.dummy_connection_data["database"]].list_collection_names.return_value = [
+            "customers"
+        ]
+        self.mock_connect.return_value[self.dummy_connection_data["database"]]["customers"].aggregate.return_value = [
+            {"cust_id": "C001", "fname": "John"}
+        ]
+
+        cust_cast = ast.TypeCast(
+            arg=ast.Identifier(parts=["customer_id"]),
+            type_name="VARCHAR",
+            precision=None,
+        )
+        cust_cast.alias = ast.Identifier(parts=["cust_id"])
+
+        fname_cast = ast.TypeCast(
+            arg=ast.Identifier(parts=["first_name"]),
+            type_name="VARCHAR",
+            precision=None,
+        )
+        fname_cast.alias = ast.Identifier(parts=["fname"])
+
+        subquery = ast.Select(
+            targets=[cust_cast, fname_cast],
+            from_table=ast.Identifier(parts=["mongo_db", "customers"]),
+            where=None,
```

Comment:
there is no subselect with 'where' in test cases, let's add at least here

Reply:
thanks, i will add that too